File: Nivedita/Array/Level2/find duplicates in an array.cpp
lass Solution{
  public:
    vector<int> duplicates(int arr[], int n) {
        vector<int> v;
        unordered_map<int,int> mp;
        for(int i=0;i<n;i++){
            mp[arr[i]]++;
        }
        
        for(auto it:mp){
            if(it.second>1){
                v.push_back(it.first);
            }
        }
        if(v.size()==0){
            v.push_back(-1);
        }
        sort(v.begin(),v.end());
        return v;
    }
};

File: JyotiKumari/Array(GFG)/Find duplicates in an array.py
class Solution:
    def duplicates(self, arr, n): 
    	# code here
    	d = {}
    	l = []
    	for element in arr:
    	    d[element]=0
    	
    	for element in arr:
    	    d[element]+=1
    	if(len(d)==n):
    	    l.append(-1)
    	else:
        	for key in d:
                if(d[key]>1):
    	            l.append(key)
    	l.sort()
    	return l 